### DISCUSSION TITLE
Help me understand how i should chose my instance?

### DISCUSSION THREAD

**Discussion by Morasithil** [searxng/searxng#2926] [Q&A] [answered]
If i understand correctly, whoever is hosting a searxng instance has the capability to log all search-requests including the useragent and ip-adress.
From what i've read anyone can host a searxng instance.
That means google, facbook, the CIA, really anyone could host an instance and keep logs of all searches.
So it would be wise to investigate which instance appears to most trustworthy as i don't have the technical knownledge to host my own instance.
So i went here https://searx.space/ and figured there are two major factors to pick an instance, country and network.
Where network stands for hosting provider?
Why i think country and provider are the most important factors is because some countrys (for example switzerland) have stronger privacy laws than others.
As for hosting providers that's a little more research as i had to look-up every single one of them to make up my mind.
To determine wheather a hosting provider is trustworthy i visited their official website and did a scan of the outbound connections that site has, so when a hosting providers website is connected to google-analytics this is a clear red flag for me and i won't use any instance that is hosted by this provider.
I struggle to understand the concept of a server being hosted in for example finland but the network (hosting provider) of that instance is located in germany, what do i make of that? What impact does this have on privacy?
Does the country where the server is physically located have the options to log all traffic or the network provider or both?

Here's a specific question about this instance https://search.projectsegfau.lt/
Country is luxembourg and network is PONYNET.
When i searched for PONYNET (using bravesearch) this was the result: https://search.brave.com/search?q=PONYNET
In short, various websites claim that PONYNET has been known to be a malware distribution center, used for spam, cybercrime and mass phishing campaigns.
Now my impression is i should not use an instance with this network provider, unless anyone here changes my mind by explaining things.

There are a few instances i was interested in but failed to find any information on their network.
https://searxng.ch/      network = NINE
https://searx.fmac.xyz/ network = THE-1984-AS
How do i find information about these networks? Atleast bravesearch doesn't seem to be capable of finding anything useful.

I've also noticed that some instances that where previously hosted by a swiss network (CH) now use (IS) iceland instead or (US) america and so on, when i chose an instance purely because of it's network provider and then after some months realize that the network provider has changed... this completely defeats the purpose of using that instance.
So i would have to manualy check the searxng instances page from time to time to make sure the network provider hasn't change?!

**Comment by Morasithil** [comment]
Anyone ??

**Reply by return42** [reply]
I do not know what answer I can give to the questions :) All the points you raised are true... but in the end you have to trust someone.  Whether an entity can be trusted cannot be answered by technical questions alone.  Especially since nobody can answer these questions completely, as long as he can only take a look from the outside. Personally, I rather trust the people who support free software or with whom I have contact.

We from the SearXNG team can only determine very general parameters, as they are also presented on searx.space.  We can't and don't want to give a recommendation for one and against the other instance.

I personally can recommend you https://searx.be, https://search.ononoki.org or https://searx.tiekoetter.com/ ... or my own instance https://darmarit.org/searx/ ... I think you can trust most of the operators who register on searx.space / most of them are committed to free software.

**Reply by dalf** [reply]
Is it correct to sum up like this: you pick an instance using the country and the provider. Since these two criteria can change without notice, you would like to be notify as soon there is a change?

If this is correct, there are different ways I can think of:
* a badge hosted on searx.space is displayed in the front page of each public instance. I'm not sure about the implication
* a bot publish the change in the instance issue. For example any change on `searx-private-search.de` of the network provider or the country is published on https://github.com/searxng/searx-instances/issues/21 --> you can subscribe to issue and be notified.

I don't know the difficulties to implement these ideas.

**Comment by unixfox** [comment, marked as answer]
First you can't really trust anyone, it depends on your threat model.

If your threat model is Google, Facebook, the CIA then you have bigger issues than finding the location of a SearXNG instance.

If you are afraid of someone logging your IP address and all your related queries then use a SearXNG instance hosted on the tor network, there are plenty of them: https://searx.space

At the end of the day, we list instances on a public list with additional details (country location for example), we try to keep only ones that work great. But it's up to the user to find which one he prefers. Whenever it is because it's close to him (load fast), located in a country that have strong privacy protection or whatever else.

You are the actor of deciding which one you want to use, if you trust no one then host one yourself using our guide: https://docs.searxng.org/. Or if you are afraid that an instance change its location then create a program to be notified, we have an API for that: https://searx.space/data/instances.json

Again, we provide the tools, but it's the user that have the final choice. We try to display as much useful information as possible, but we can't cover the case/needs of everyone I'm sorry.

**Reply by Morasithil** [reply]
Thanks for the reply.

While the CIA is not a threat model for me, google and facebook are (bigtech in general).
I don't like them harvesting my data like a mad stalker, that's all.
I'm using a VPN (mullvad) over Tor, but i will think about that suggestion. Tor relays could also be hosted by the CIA just saying.

Thing is, if i just want to use searxng without thinking then i would randomly select an instance and use it, but i preffer doing atleast the barbones minimal research to make sure the instance i'm using should be trustworthy.
I still don't know how to chose an instance but i won't further consume any of your time as you made clear in your last reply you can't fit everyones needs.
I'll have a look at the tools but as i'm not a programmer my hopes aren't high.

Thank you, have a good time.
